WS2012R2/lisa/setupscripts/RemoveIsoFromDvd.ps1

```diff
@@ -87,13 +88,30 @@ $error.Clear()
 #
 # Remove the .iso file from the VMs DVD drive
 #
+
+#
+#Get the version of VM
+#
+$vmGeneration = Get-VM $vmName | select -ExpandProperty Generation
+#
+# Make sure the DVD drive exists on the VM
+#
+if ($vmGeneration -eq 1)
+{
+    $controllerNumber=1
+}
+else
+{
+    $controllerNumber=0
+}
+
 $dvdcount = $(Get-VMDvdDrive -VMName $vmName).ControllerLocation.count 
-for ($i=0; $i -lt $dvdcount; $i++)
+for ($i=0; $i -le $dvdcount; $i++)
 {
-    $dvd = Get-VMDvdDrive -VMName $vmName -ControllerNumber 1 -ControllerLocation $i
+    $dvd = Get-VMDvdDrive -VMName $vmName -ControllerNumber $controllerNumber -ControllerLocation $i
```

Comment:
Same here for using -ComputerName, I see that in the older code it wasn't either specified.